### ISSUE TITLE
`prefer-package-imports` error thrown for files that are ingored in .regal/config.yaml

### ISSUE THREAD

**Issue by devin-brenton** [StyraInc/regal#605]
Given a config.yaml with:
```
ignore:
  files:
    - assert.rego
```

Where assert.rego is defined as:

```
package test.assert

...
```

I'm receiving the following regal error:
![image](https://github.com/StyraInc/regal/assets/17804106/3d44a078-e87f-4b69-b9d6-f8951208f539)

If I remove the ignored file from config.yaml the error goes away.

**Comment by anderseknert** [comment]
Hi Devin! And thanks for reporting this. 

You're absolutely right about this being an issue, and it relates to this being an "aggregate rule" (see bottom of [this section](https://github.com/styrainc/regal?tab=readme-ov-file#rules) in the README for details), for which we [don't yet](https://github.com/StyraInc/regal/issues/344) have a good way to handle ignore directives. I think we might want some way to say "ignore this file when _aggregating_ data", at some point, but a good start would be to just use ignore directives for these rules the same way as we do with other rules. I'll take a look at this next week.

**Comment by anderseknert** [comment]
The unresolved imports issue fixed in #607

While this doesn't fully address the issue of ignore directives in aggregate rules, I *think* this fixes the issue you're seeing, as the file you ignored wasn't read and the error then was that the import was unresolved. The larger issue is that we'll need to think through what it means to ignore a file for the purpose of aggregation. Does it mean that:

1. We should still collect data from ignored files, but not report errors found in the files themselves?
2. We should not collect data from ignored files, and not report errrors found in the files?

I'm leaning towards option 1, as you might not want the linter to report issues in e.g. tests, but still aggregate data from these to do things like determine what rules are unused in your non-tests files. I guess the most diplomatic option is to add another config option that'll let one toggle whether to include ignored files in data aggregation. Oh well, we'll get to that eventually. 

> This package doesn't exist, yet the rule is being thrown. This type of error is already handled by the OPA engine and there is no need for a lint error as well.

No, [it's not handled by OPA](https://github.com/open-policy-agent/opa/issues/491), or at least not in a way many would expect (i.e. unresolved imports are simply undefined). This has been a source of frustration to many, and something we intend to have [Regal help flag](https://github.com/StyraInc/regal/issues/300) eventually. That should however be a rule of its own, and flagged independently of rules likes this one.

Anyway, if you could build Regal from this PR branch and test it out for your issue, that'd be much appreciated!

**Comment by devin-brenton** [comment]
Thanks for the quick turnaround! Do you mind pointing me in the right direction for how to update my install to test this? I'm currently running:
`curl -L -o /usr/local/bin/regal https://github.com/StyraInc/regal/releases/latest/download/regal_Linux_${PLATFORM}`

**Comment by anderseknert** [comment]
I just merged the PR, so no you can at least skip the `gh pr checkout` step and just build from `main` after cloning 😅

**Comment by devin-brenton** [comment]
That change seems to do the trick! Thanks

**Comment by anderseknert** [comment]
Awesome! Thanks for reporting back 👍 I'll close this issue then, with related issues for ignore directives for aggregate rules being addressed in #344 

If you encounter any other issues, let us know!
